## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#ded79d0b-6144-4f5f-91db-f13158abce6c) |
| mint-large-bucket.sh | :x: [more...](#bc6743c3-ded8-4999-a87a-3dfad4475f81) |
<a name="bc6743c3-ded8-4999-a87a-3dfad4475f81"></a>
### 8286-39968aa/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:31155
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 61d6721b2961:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 5 seconds
(3/14) Running aws-sdk-php tests ... done in 54 seconds
(4/14) Running aws-sdk-ruby tests ... done in 21 seconds
(5/14) Running awscli tests ... done in 2 minutes and 24 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 23 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 24 seconds
(9/14) Running minio-go tests ... done in 2 minutes and 27 seconds
(10/14) Running minio-java tests ... done in 3 minutes and 13 seconds
(11/14) Running minio-js tests ... done in 1 minutes and 5 seconds
(12/14) Running minio-py tests ... FAILED in 6 minutes and 10 seconds
{
  "status": "FAIL",
  "duration": 667,
  "args": {
    "response.reason": "Forbidden",
    "object_name": "1746a9f9-272d-4da7-b320-1d9014d4401f",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-09-22T10-16-34Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15C6BD02AFB61898\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Sun, 22 Sep 2019 10:48:59 GMT\"]}}",
    "bucket_name": "minio-py-test-5e963aec-0a11-4a32-a128-4d3d2fa203cb",
    "presigned_get_object_url": "http://72.28.97.52:31155/minio-py-test-5e963aec-0a11-4a32-a128-4d3d2fa203cb/1746a9f9-272d-4da7-b320-1d9014d4401f?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20190922%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20190922T104853Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=169f836867ddbc03e64cd7e04a63c9202a85b2d1fb545b5bf5d382f1c67da527",
    "response.status": 403,
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>1746a9f9-272d-4da7-b320-1d9014d4401f</Key><BucketName>minio-py-test-5e963aec-0a11-4a32-a128-4d3d2fa203cb</BucketName><Resource>/minio-py-test-5e963aec-0a11-4a32-a128-4d3d2fa203cb/1746a9f9-272d-4da7-b320-1d9014d4401f</Resource><RequestId>15C6BD02AFB61898</RequestId><HostId>fee64c3f-25b3-4974-bf50-9f21b76a1223</HostId></Error>"
  },
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "message": "AccessDenied: message: Access Denied",
  "name": "minio-py:test_presigned_get_object_expiry_5sec"
}

Executed 11 out of 14 tests successfully.
```
<a name="ded79d0b-6144-4f5f-91db-f13158abce6c"></a>
### 8286-39968aa/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:31038
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4ca86d972e4a:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-fdrv6xgg9pq3b9",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 365400,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15C6BC4668C85133, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

--- a/cmd/posix.go
+++ b/cmd/posix.go
@@ -329,6 +329,10 @@ func (s *posix) DiskInfo() (info DiskInfo, err error) {
 		return info, errFaultyDisk
 	}
 
+	if err := s.checkDiskFound(); err != nil {
+		return info, err
+	}
+
 	di, err := getDiskInfo(s.diskPath)
 	if err != nil {
 		return info, err

